File: Web/src/app/rx/inner-mq/client/inner-mq.client.ts
import { Subject } from "rxjs";
import { Topic } from "../topic";
import { PersistentType } from "../service/inner-mq.service";

export interface InnerMqClient {

	getId(): string;

	getSubject(topic: Topic): Subject<any> | undefined;

	isDestroyed(): boolean;

	sub<T>(topic: Topic, subscribe: (e: T) => void): string;

	pub(topic: Topic, msg: any, option?: { persistent: boolean, type: PersistentType }): void;

	stopSubByTopic(topic: Topic): void;

	stopSubBySubscriptionId(id: string): void;

	destroy(): void;

}

File: Code/src/main/java/com/jmd/rx/client/InnerMqClient.java
package com.jmd.rx.client;

import com.jmd.rx.Topic;
import com.jmd.rx.callback.OnMessageCallback;
import com.jmd.rx.callback.OnSubscribeCallback;
import io.reactivex.rxjava3.subjects.PublishSubject;

public interface InnerMqClient {

    void setOnSubscribeCallback(OnSubscribeCallback onSubscribeCallback);

    String getId();

    PublishSubject<Object> getSubject(Topic topic);

    boolean isDestroyed();

    <T> void sub(Topic topic, OnMessageCallback<T> callback);

    void destroy();

}
